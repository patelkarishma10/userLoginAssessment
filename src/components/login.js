import React, { Component } from 'react';
import axios from "axios";

export class Login extends Component {


    constructor() {
        super();
        this.state = {
            message: ""
        };
    }

//  {password: password}

    login = (e) => {
        e.preventDefault();
        let username = e.target[0].value;
        let password = e.target[1].value;
         if (username === "" || password === "" ) {
        this.setState({ message: "Please fill all fields" })
    } else {
        axios
            .get("http://localhost:5000/user/name/"+ username + "/" + password )
            .then(response => {
                this.setState({ message:  JSON.stringify(response.data.Status) });
                console.log(response.data);
            })
            
    }
    }

    render() {

        return (
            <div className="formDiv">
                <h1>Login</h1>
                <form onSubmit={this.login}>
                    <label htmlFor="username">Username: </label>
                    <input type="text" id="username" className="form-control" />
                    <br />

                    <label htmlFor="password" >Password: </label>
                    <input type="content" id="password" className="form-control" />
                    <br />

                    <input type="submit" className="btn btn-dark" value="login" />
                </form>

                 <p style={{color: 'red'}}>{this.state.message}</p> 
            </div>

        );
    }
}