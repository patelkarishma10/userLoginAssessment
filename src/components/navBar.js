import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

export class NavBar extends Component {
render() {
        return (
          <div>
              <ul className="navBar">
                <li id="Register" ><Link to="/">Register</Link></li>
                <li id="Login"><Link to="/login">Login</Link></li>
                <li id="See all Users"><Link to="/userList">See all users</Link></li>
              </ul>
          </div>
        )
      }
}