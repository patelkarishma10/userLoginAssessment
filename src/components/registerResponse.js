import React, { Component } from 'react';

export class RegisterResponse extends Component {
    render(props) {
        return (
            <tbody>
                <tr><td>{this.props.responseData.username}</td></tr>
                <tr><td>{this.props.responseData.email}</td></tr>
                <tr><td>{this.props.responseData.password}</td></tr>
                <tr><td>{this.props.responseData.password2}</td></tr>
            </tbody>
        );
    }
}