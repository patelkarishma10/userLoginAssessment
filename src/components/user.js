import React, { Component } from 'react';

export class User extends Component {
    render(props) {

const { 
    username
} = this.props

        return (

                        <tr>
                            <td>{username}</td>
                        </tr>


        );
    }
}