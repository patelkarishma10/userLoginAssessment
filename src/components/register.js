import React, { Component } from 'react';
import axios from "axios";
import { Table } from "reactstrap";

import { RegisterResponse } from './registerResponse';

export class Register extends Component {

    constructor() {
        super();
        this.state = {
            responseData: ""
        };
    }

    register = (e) => {
        e.preventDefault();
        let newItem = {
            username: e.target[0].value,
            email: e.target[1].value,
            password: e.target[2].value,
            password2: e.target[3].value
        }

        axios
            .post("http://localhost:5000/user/createUser", newItem)

            .then(response => {
                this.setState({
                    responseData: (response.data)
                });
                this.props.passedFunction();
            });


    }

    render() {

        return (
            <div className="formDiv">
                <h1>Register</h1>
                <form onSubmit={this.register}>
                    <label htmlFor="username">Username: </label>
                    <input type="text" id="username" className="form-control" />
                    <br />

                    <label htmlFor="email" >Email: </label>
                    <input type="text" id="email" className="form-control" />
                    <br />

                    <label htmlFor="password" >Password: </label>
                    <input type="password" id="password" className="form-control" />
                    <br />

                    <label htmlFor="password2" >password2: </label>
                    <input type="password" id="password2" className="form-control" />
                    <br />

                    <input type="submit" className="btn btn-dark" value="register" />
                </form>


                <Table style={{ color: 'red' }}>
                    <RegisterResponse responseData={this.state.responseData} />
                </Table>
            </div>

        );
    }
}