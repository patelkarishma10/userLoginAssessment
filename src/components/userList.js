import React, { Component } from 'react';
import { Table } from "reactstrap";

import { User } from './user';

export class UserList extends Component {

    render() {

        return (
            <div className="formDiv">
                <div id="random"> </div>
                <h2>Users</h2>
                <Table className="formDiv">
                    <tbody>

                        {this.props.dataSent.map((item) => (

                            <User key={item._id}
                                
                                username={item.username} />

                        ))}

                    </tbody>
                </Table>
            </div>

        );
    }
}