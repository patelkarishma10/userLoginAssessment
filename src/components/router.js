import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

import { Register } from './register';
import { Login } from './login';
import { UserList } from './userList';
import { NavBar } from './navBar';

export class RouterComponent extends Component {
    render() {
        return (
            <div>
                <Router>
                    <NavBar />
                    <Route exact path="/" render={(props) => <Register passedFunction={this.props.onLoad} />} />
                    {/* <Route path="/login" render={(props) => <Login passedFunction={this.props.onLoad} />} /> */}
                    <Route path="/login" component={Login} />
                    <Route path="/userList" render={(props) => <UserList passedFunction={this.props.onLoad} dataSent={this.props.dataSent} />} />
                </Router>
            </div>
        )
    }
}
