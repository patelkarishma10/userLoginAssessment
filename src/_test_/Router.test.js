import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'


import {RouterComponent} from '../components/router';
import {NavBar} from '../components/navBar';
import { Login } from '../components/login';
import { UserList } from '../components/userList';
import { Register } from '../components/register';

Enzyme.configure({ adapter: new Adapter() })

it('snapshot router', () => {

  const tree = renderer.create(<RouterComponent/>).toJSON()
  expect(tree).toMatchSnapshot();
})

it("render a prop to login", () => {
    const wrapper = shallow(<Login passedFunction="[]" />);
    expect(wrapper.instance().props.passedFunction).toEqual("[]");
    expect(wrapper).toMatchSnapshot();
  });


  it("render a prop to register", () => {
    const wrapper = shallow(<Register passedFunction="[]" />);
    expect(wrapper.instance().props.passedFunction).toEqual("[]");
    expect(wrapper).toMatchSnapshot();
  });
  
      it("render a prop", () => {
    const wrapper = shallow(<UserList dataSent={[{username: ""}, {username: "test"}]} />);
    expect(wrapper.instance().props.dataSent).toEqual([{username: ""}, {username: "test"}]);
    expect(wrapper).toMatchSnapshot();
  });
