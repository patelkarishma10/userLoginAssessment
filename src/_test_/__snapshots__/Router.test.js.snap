// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a prop 1`] = `ShallowWrapper {}`;

exports[`render a prop to register 1`] = `ShallowWrapper {}`;

exports[`snapshot router 1`] = `
<div>
  <div>
    <ul
      className="navBar"
    >
      <li
        id="Register"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <li
        id="Login"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li
        id="See all Users"
      >
        <a
          href="/userList"
          onClick={[Function]}
        >
          See all users
        </a>
      </li>
    </ul>
  </div>
  <div
    className="formDiv"
  >
    <h1>
      Register
    </h1>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        Username: 
      </label>
      <input
        className="form-control"
        id="username"
        type="text"
      />
      <br />
      <label
        htmlFor="email"
      >
        Email: 
      </label>
      <input
        className="form-control"
        id="email"
        type="text"
      />
      <br />
      <label
        htmlFor="password"
      >
        Password: 
      </label>
      <input
        className="form-control"
        id="password"
        type="text"
      />
      <br />
      <label
        htmlFor="password2"
      >
        password2: 
      </label>
      <input
        className="form-control"
        id="password2"
        type="text"
      />
      <br />
      <input
        className="btn btn-dark"
        type="submit"
        value="register"
      />
    </form>
    <table
      className="table"
      style={
        Object {
          "color": "red",
        }
      }
    >
      <tbody>
        <tr>
          <td />
        </tr>
        <tr>
          <td />
        </tr>
        <tr>
          <td />
        </tr>
        <tr>
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
